staging/src/k8s.io/kubectl/pkg/generate/versioned/run.go

```diff
@@ -178,14 +178,14 @@ func updatePodPorts(params map[string]string, podSpec *v1.PodSpec) (err error) {
 	port := -1
 	hostPort := -1
 	if len(params["port"]) > 0 {
-		port, err = strconv.Atoi(params["port"])
+		port, err = utilsnet.ParsePort(params["port"], false)
 		if err != nil {
 			return err
 		}
 	}
 
 	if len(params["hostport"]) > 0 {
-		hostPort, err = strconv.Atoi(params["hostport"])
+		hostPort, err = utilsnet.ParsePort(params["port"], false)
```

Comment:
this typos the param, should be `hostport`

Reply:
that also means there are no tests exercising the hostport param, which is not good

Reply:
a hostport param of 0 was previously tolerated as a no-op, right?

Reply:
Sry, I ignored it, it has been corrected

Reply:
> a hostport param of 0 was previously tolerated as a no-op, right?

yes, it should be `true`.